aws/resource_aws_iam_role_test.go

```diff
@@ -491,321 +494,1411 @@ func TestAccAWSIAMRole_tags(t *testing.T) {
+		r, ok := s.RootModule().Resources[resource]
+		if !ok {
+			return fmt.Errorf("Resource not found")
+		}
+		name, ok := r.Primary.Attributes["name"]
+		if !ok {
+			return fmt.Errorf("Name attr not found: %#v", r.Primary.Attributes)
+		}
+		if !strings.HasPrefix(name, prefix) {
+			return fmt.Errorf("Name: %q, does not have prefix: %q", name, prefix)
+		}
+		return nil
+	}
+}
+
+// Attach inline policy out of band (outside of terraform)
+func testAccAddAwsIAMRolePolicy(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Resource not found")
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Role name is set")
+		}
+
+		iamconn := testAccProvider.Meta().(*AWSClient).iamconn
+
+		input := &iam.PutRolePolicyInput{
 			RoleName: aws.String(rs.Primary.ID),
 			PolicyDocument: aws.String(`{
   "Version": "2012-10-17",
-  "Statement": {
-    "Effect": "Allow",
-    "Action": "*",
-    "Resource": "*"
-  }
-}`),
-			PolicyName: aws.String(resource.UniqueId()),
-		}
+  "Statement": {
+    "Effect": "Allow",
+    "Action": "*",
+    "Resource": "*"
+  }
+}`),
+			PolicyName: aws.String(resource.UniqueId()),
+		}
+
+		_, err := iamconn.PutRolePolicy(input)
+		return err
+	}
+}
+
+func testAccCheckAWSRolePermissionsBoundary(getRoleOutput *iam.GetRoleOutput, expectedPermissionsBoundaryArn string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		actualPermissionsBoundaryArn := ""
+
+		if getRoleOutput.Role.PermissionsBoundary != nil {
+			actualPermissionsBoundaryArn = *getRoleOutput.Role.PermissionsBoundary.PermissionsBoundaryArn
+		}
+
+		if actualPermissionsBoundaryArn != expectedPermissionsBoundaryArn {
+			return fmt.Errorf("PermissionsBoundary: '%q', expected '%q'.", actualPermissionsBoundaryArn, expectedPermissionsBoundaryArn)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSRolePolicyCheckInline(role *iam.GetRoleOutput, roleName string, inlinePolicies []string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if !strings.Contains(*role.Role.RoleName, roleName) {
+			return fmt.Errorf("bad role: expected %s, got %s", roleName, *role.Role.RoleName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).iamconn
+
+		policyNames, err := readIamRolePolicyNames(conn, roleName)
+		if err != nil {
+			return fmt.Errorf("reading role (%s): %w", roleName, err)
+		}
+
+		var inlinePolicyList []string
+		for _, name := range policyNames {
+			inlinePolicyList = append(inlinePolicyList, aws.StringValue(name))
+		}
+
+		if !compareStringSlices(inlinePolicyList, inlinePolicies) {
+			return fmt.Errorf("inline policies did not match: %s (from AWS) to %s (expected)", strings.Join(inlinePolicyList, ","), strings.Join(inlinePolicies, ","))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSRolePolicyCheckManaged(role *iam.GetRoleOutput, roleName string, managedPolicies []string) resource.TestCheckFunc {
```

Comment:
Similarly here, this can be replaced with [`resource.TestCheckTypeSetElemAttr`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCheckTypeSetElemAttr) or [`resource.TestCheckTypeSetElemAttrPair`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCheckTypeSetElemAttrPair) or removed entirely due to `ImportStateVerify` testing.

Reply:
Removed entirely